nerfstudio/data/pixel_samplers.py

```diff
@@ -48,30 +48,19 @@ def collate_image_dataset_batch(batch: Dict, num_rays_per_batch: int, keep_full_
         ).long()
 
     c, y, x = (i.flatten() for i in torch.split(indices, 1, dim=-1))
-    image = batch["image"][c, y, x]
-    mask, semantics_stuff, semantics_thing = None, None, None
-    if "mask" in batch:
-        mask = batch["mask"][c, y, x]
-    if "semantics_stuff" in batch:
-        semantics_stuff = batch["semantics_stuff"][c, y, x]
-    if "semantics_thing" in batch:
-        semantics_thing = batch["semantics_thing"][c, y, x]
-    assert image.shape == (num_rays_per_batch, 3), image.shape
+    collated_batch = {key: value[c, y, x] for key, value in batch.items() if key != "image_idx" and value is not None}
+
+    assert collated_batch["image"].shape == (num_rays_per_batch, 3), collated_batch["image"].shape
 
     # Needed to correct the random indices to their actual camera idx locations.
     local_indices = indices.clone()
     indices[:, 0] = batch["image_idx"][c]
-    collated_batch = {
-        "local_indices": local_indices,  # local to the batch returned
-        "indices": indices,  # with the abs camera indices
-        "image": image,
-    }
-    if mask is not None:
-        collated_batch["mask"] = mask
-    if semantics_stuff is not None:
-        collated_batch["semantics_stuff"] = semantics_stuff
-    if semantics_thing is not None:
-        collated_batch["semantics_thing"] = semantics_thing
+    collated_batch.update(
+        {
+            "local_indices": local_indices,  # local to the batch returned
```

Comment:
I guess we should really just delete this. It's not used anywhere in the code anymore. We used it previously, but it's unnecessary.